archive/progress_http/src/test/scala/uk/ac/wellcome/platform/archive/progress_http/ProgressHttpFeatureTest.scala

```diff
@@ -216,5 +285,353 @@ class ProgressHttpFeatureTest
+                 |  }
+                 |}""".stripMargin
+            )
+
+            val request = HttpRequest(
+              method = POST,
+              uri = url,
+              headers = Nil,
+              entity = entity
+            )
+
+            whenRequestReady(request) { response: HttpResponse =>
+              response.status shouldBe StatusCodes.BadRequest
+              response.entity.contentType shouldBe ContentTypes.`application/json`
+
+              val progressFuture =
+                Unmarshal(response.entity).to[ErrorResponse]
+
+              whenReady(progressFuture) { actualError =>
+                actualError shouldBe ErrorResponse(
+                  400,
+                  "Invalid value at .sourceLocation.provider.id: invalid value supplied, valid values are: aws-s3-standard, aws-s3-ia.",
+                  "Bad Request",
+                  "Error")
+                val requests =
+                  listMessagesReceivedFromSNS(topic).map(messageInfo =>
+                    fromJson[IngestBagRequest](messageInfo.message).get)
+
+                requests shouldBe empty
+              }
+            }
+          }
+      }
+    }
+
+    it("returns a json error if the ingestType doesn't have a valid id field") {
+      withConfiguredApp {
+        case (_, topic, baseUrl) =>
+          withMaterializer { implicit materialiser =>
+            val url = s"$baseUrl/progress"
+
+            val entity = HttpEntity(
+              ContentTypes.`application/json`,
+              """|{
+                 |  "type": "Ingest",
+                 |  "ingestType": {
+                 |    "id": "baboop",
+                 |    "type": "IngestType"
+                 |  },
+                 |  "sourceLocation":{
+                 |    "type": "Location",
+                 |    "provider": {
+                 |      "type": "Provider",
+                 |      "id": "aws-s3-standard"
+                 |    },
+                 |    "bucket": "bucket",
+                 |    "path": "b22454408.zip"
+                 |  },
+                 |  "space": {
+                 |    "id": "bcnfgh",
+                 |    "type": "Space"
+                 |  }
+                 |}""".stripMargin
+            )
+
+            val request = HttpRequest(
+              method = POST,
+              uri = url,
+              headers = Nil,
+              entity = entity
+            )
+
+            whenRequestReady(request) { response: HttpResponse =>
+              response.status shouldBe StatusCodes.BadRequest
+              response.entity.contentType shouldBe ContentTypes.`application/json`
+
+              val progressFuture =
+                Unmarshal(response.entity).to[ErrorResponse]
+
+              whenReady(progressFuture) { actualError =>
+                actualError shouldBe ErrorResponse(
+                  400,
+                  "Invalid value at .ingestType.id: invalid value supplied, valid values are: create.",
```

Comment:
“Invalid value at, invalid value supplied”? Bit repetitive. How about:

> Invalid value at .ingestType.id: got "baboop", expected "create".

Reply:
Yeah, you're right, it does repeat. I'll change it. How about
```Invalid value at .ingestType.id: got "baboop", valid values are: create.```
This way we can change the rejectionhandler in  `Router` to use "valid values" in the match block